### ISSUE TITLE
NameError: name 'PY3' is not defined

### ISSUE THREAD

**Issue by MSusik** [catboost/catboost#269]
Hi!

I installed `catboost` on Python 3.6.3 within a Docker container (python:3.6 image). The installation was done through PyPI. The library version is 0.6.3

I can import catboost, but when I do:

`model = catboost.CatBoostRegressor()`

I obtain an error:

```
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-16-e4bd7d18c439> in <module>()
----> 1 model = catboost.CatBoostRegressor()

/usr/local/lib/python3.6/site-packages/catboost/core.py in __init__(self, iterations, learning_rate, depth, l2_leaf_reg, model_size_reg, rsm, loss_function, border_count, feature_border_type, fold_permutation_block_size, od_pval, od_wait, od_type, nan_mode, counter_calc_method, leaf_estimation_iterations, leaf_estimation_method, thread_count, random_seed, use_best_model, verbose, logging_level, metric_period, ctr_leaf_count_limit, store_all_simple_ctr, max_ctr_complexity, has_time, one_hot_max_size, random_strength, name, ignored_features, train_dir, custom_metric, eval_metric, bagging_temperature, save_snapshot, snapshot_file, fold_len_multiplier, used_ram_limit, gpu_ram_part, allow_writing_files, approx_on_full_history, boosting_type, simple_ctr, combinations_ctr, per_feature_ctr, ctr_description, task_type, device_config, devices, bootstrap_type, max_depth, n_estimators, num_boost_round, num_trees, colsample_bylevel, random_state, reg_lambda, objective, eta, max_bin, **kwargs)
   1767             objective = None
   1768 
-> 1769         if isinstance(loss_function, str) and self._is_classification_loss(loss_function):
   1770             raise CatboostError("Invalid loss_function={}: for Regressor use RMSE, MAE, Quantile, LogLinQuantile, Poisson, MAPE, R2.".format(loss_function))
   1771         params = {}

_catboost.pyx in _catboost._CatBoostBase._is_classification_loss()

_catboost.pyx in _catboost.to_binary_str()

NameError: name 'PY3' is not defined

```

Same happens when I specify a loss function:

`model = catboost.CatBoostRegressor(loss_function='MAPE')`

**Comment by aldanor** [comment]
Probably caused by https://github.com/catboost/catboost/issues/268 - try `import catboost._catboost` and see what happens.

**Comment by zhangruinan** [comment]
same problem here

**Comment by aldanor** [comment]
See the comment above

**Comment by zhangruinan** [comment]
Tried, still error. 

--------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-4-aa08a313613e> in <module>()
----> 1 import catboost._catboost

_catboost.pyx in init _catboost()

AttributeError: type object '_catboost._FloatArrayWrapper' has no attribute '__reduce_cython__'

**Comment by KelvinZx** [comment]
same problem, i tried import catboost._catboost and get message below

---------------------------------------------------------------------------
      1 from catboost import CatBoostRegressor
      2 import numpy
----> 3 import catboost._catboost

_catboost.pyx in init _catboost()

AttributeError: type object '_catboost._FloatArrayWrapper' has no attribute '__reduce_cython__'

**Comment by esceptico** [comment]
try to 'pip install PY3'
worked for me

**Comment by esceptico** [comment]
wow, on my home pc my previous answer doesn't work

**Comment by sharthZ** [comment]
Same problem, nothing above didnt work for me

**Comment by annaveronika** [comment]
The fix is on github, could you please check if it works now for you? To do that you need to build from source (https://tech.yandex.com/catboost/doc/dg/concepts/python-installation-docpage/)

**Comment by StrausMG** [comment]
+1
That's how I faced the same problem:

```
features  = <some dataframe>
target = <some series>
p = catboost.Pool(features, target)
```

```
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-224-a3edbf41d908> in <module>()
----> 1 catboost.Pool(f, t)

~/miniconda3/envs/tafd/lib/python3.6/site-packages/catboost/core.py in __init__(self, data, label, cat_features, column_description, pairs, delimiter, has_header, weight, group_id, subgroup_id, pairs_weight, baseline, feature_names, thread_count)
    206                 self._read(data, column_description, pairs, delimiter, has_header, thread_count)
    207             else:
--> 208                 self._init(data, label, cat_features, pairs, weight, group_id, subgroup_id, pairs_weight, baseline, feature_names)
    209         super(Pool, self).__init__()
    210 

~/miniconda3/envs/tafd/lib/python3.6/site-packages/catboost/core.py in _init(self, data_matrix, label, cat_features, pairs, weight, group_id, subgroup_id, pairs_weight, baseline, feature_names)
    518         if feature_names is not None:
    519             self._check_feature_names(feature_names, features_count)
--> 520         self._init_pool(data_matrix, label, cat_features, pairs, weight, group_id, subgroup_id, pairs_weight, baseline, feature_names)
    521 
    522 

_catboost.pyx in _catboost._PoolBase._init_pool()

_catboost.pyx in _catboost._PoolBase._init_pool()

_catboost.pyx in _catboost._PoolBase._set_feature_names()

_catboost.pyx in _catboost.to_binary_str()

NameError: name 'PY3' is not defined
```

Ubuntu 16.04
All the software (python 3.6.5, numpy, six, catboost, whatever else) is of the latest versions currently available.

**Comment by jrzaurin** [comment]
in my case appears for a different reason, not at `_set_feature_names`

```
clf = ctb.CatBoostClassifier(eval_metric='F1', scale_pos_weight=9.)
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-9-15bc93ece38c> in <module>()
----> 1 clf = ctb.CatBoostClassifier(eval_metric='F1', scale_pos_weight=9)

~/.virtualenvs/py3/lib/python3.6/site-packages/catboost/core.py in __init__(self, iterations, learning_rate, depth, l2_leaf_reg, model_size_reg, rsm, loss_function, border_count, feature_border_type, fold_permutation_block_size, od_pval, od_wait, od_type, nan_mode, counter_calc_method, leaf_estimation_iterations, leaf_estimation_method, thread_count, random_seed, use_best_model, verbose, logging_level, metric_period, ctr_leaf_count_limit, store_all_simple_ctr, max_ctr_complexity, has_time, classes_count, class_weights, one_hot_max_size, random_strength, name, ignored_features, train_dir, custom_loss, custom_metric, eval_metric, bagging_temperature, save_snapshot, snapshot_file, fold_len_multiplier, used_ram_limit, gpu_ram_part, allow_writing_files, approx_on_full_history, boosting_type, simple_ctr, combinations_ctr, per_feature_ctr, ctr_description, task_type, device_config, devices, bootstrap_type, max_depth, n_estimators, num_boost_round, num_trees, colsample_bylevel, random_state, reg_lambda, objective, eta, max_bin, scale_pos_weight, **kwargs)
   1548             objective = None
   1549
-> 1550         if isinstance(loss_function, str) and not self._is_classification_loss(loss_function):
   1551             raise CatboostError("Invalid loss_function='{}': for classifier use "
   1552                                 "Logloss, CrossEntropy, MultiClass, MultiClassOneVsAll, AUC, Accuracy, Precision, Recall, F1, TotalF1, MCC or custom objective object".format(loss_function))

_catboost.pyx in _catboost._CatBoostBase._is_classification_loss()

_catboost.pyx in _catboost.to_binary_str()

NameError: name 'PY3' is not defined
```

I am on CentOS7, Python 3.6.4 and updated versions for everything else

**Comment by aldanor** [comment]
Guys, as I’ve already replied above, this error is the consequence (of the extension not being fully imported) and not the underlying cause, see #268.

**Comment by jrzaurin** [comment]
Hey @aldanor 

ok, let me check there. Sorry for the repetition

**Comment by shaz13** [comment]
Same issue here

**Comment by jrzaurin** [comment]
@aldanor already replied up there that _"this error is the consequence (of the extension not being fully imported) and not the underlying cause, see #268."_

Nonetheless, in case is of any help, for me this works:

```
from catboost import CatBoost
import catboost as ctb
```

and off we go, from there in advance all works smoothly

hope it helps

**Comment by kizill** [comment]
We fixed this bug in master, our plan is to ship new package version next tuesday.

**Comment by aldanor** [comment]
@kizill Thanks; are you also planning to finally push the conda package with the fixed master?

**Comment by kizill** [comment]
@aldanor yes, we hope it will work:)

**Comment by sharthZ** [comment]
@kizill gj! Is it possible to install with pip fixed version of python-package from git?

**Comment by kizill** [comment]
We released 0.8 with this fix, can someone try it?

**Comment by esceptico** [comment]
@kizill Yes, everything works perfectly, thank you
